packages/serenity-bdd/package.json

```diff
@@ -38,7 +38,8 @@
     "clean": "rimraf target",
     "test": "c8 npm run test:no-coverage",
     "test:no-coverage": "mocha --config ../../.mocharc.yml 'spec/**/*.spec.*'",
-    "compile": "rimraf lib && tsc --project tsconfig.build.json"
+    "compile": "rimraf lib cache && tsc --project tsconfig.build.json && npm run populate-cache",
+    "populate-cache": "./bin/serenity-bdd update --cacheDir './cache'"
```

Comment:
By the way, if you'd like to commit these changes, please feel free to mark them as `fix` or `chore`; your initial `feat` commit will "win" when the CI pipeline determines the build number and bump the `minor` version.

Reply:
> If you renamed populate-cache to postcompile, NPM would automatically call this [post-script](https://docs.npmjs.com/cli/v9/using-npm/scripts#pre--post-scripts) when compile passes.

Hmm, this actually looks like a more ergonomic way, I'll try it and probably will stick with the `post` hook so we can avoid introduction of the additional explicit step in the CI.
By the way, `pre` and `post` hooks are stable in the npm, however they're not supported by [yarn](https://yarnpkg.com/advanced/lifecycle-scripts) and was disabled by default in [pnpm](https://github.com/pnpm/pnpm/issues/2891) (although reverted back in [pnpm v9](https://github.com/pnpm/pnpm/pull/7634)). I think it's not a problem for SerenityJS as it's using npm, but might be useful if you'll ever consider to switch the package manager for the project.

> I'd keep the --ignoreSSL flag for contributors whose network setups struggle with correct handling of SSL certificates. Unfortunatelly, there's still plenty of broken corporate networks out there.

Nice catch, didn't considered this case. Will add this flag.

> I'd add node to make the interpreter explicit.

I've decided to not use `node` as it was my assumption that it works that way when executed via CLI by the user. In that case, it looks like an implicit integration test. But I don't insist on that as I'm actually not sure in the correctness of my assumption and I can change this script to call `node` explicitly to make it more reliable.

Reply:
> By the way, pre and post hooks are stable in the npm, however they're not supported by yarn and was disabled by default in pnpm (although reverted back in pnpm v9). I think it's not a problem for SerenityJS as it's using npm, but might be useful if you'll ever consider to switch the package manager for the project.

Oh, that's a good tip, thank you. I wanted to investigate using pnpm instead of the regular npm since new versions of nx (replacement of lerna) don't support hoisting or workspaces. I'll keep your advice in mind.

> I've decided to not use node as it was my assumption that it works that way when executed via CLI by the user.

Windows-based machines might sometimes get confused, adding an explicit call to `node` helps to avoid that.